### PR TITLE
HPCC-29032 Update default cmake options

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

File: cmake_modules/plugins.cmake [hunk 1]
```diff
@@ -16,7 +16,6 @@ MACRO(SET_PLUGIN_PACKAGE plugin)
     set(PLATFORM OFF)
     set(INCLUDE_PLUGINS OFF)
     set(SIGN_MODULES OFF)
-    set(USE_OPTIONAL OFF) # Force failure if we can't find the plugin dependencies
```
Comment (GordonSmith) [inline cmake_modules/plugins.cmake:19]: Should keep this...

**Review by ghalliday — COMMENTED** [review]
The changes look sensible.  One question.  Also, the jira should contain details of the changes that were made - e.g.. always enabling icu, changing the defaults for X Y and Z, for documentation and to make the review easier.

File: cmake_modules/commonSetup.cmake [hunk 2]
```diff
@@ -830,43 +806,23 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
...
-          if (Boost_REGEX_FOUND)
-            message(STATUS "BOOST_REGEX enabled")
-            add_definitions (-D_USE_BOOST_REGEX)
-          else()
-            message(FATAL_ERROR "BOOST_REGEX requested but package not found")
-          endif()
+          message(FATAL_ERROR "BOOST_REGEX requested but package not found")
         endif()
       else(USE_BOOST_REGEX)
-        if (USE_C11_REGEX)
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply cmake_modules/commonSetup.cmake:847]: Probably to provoke discussion - it looked like it was only there for CentOS 6/7 support as boost wasn't an option?

I would remove the "option" for regex and just include it?

[the remark this answers is not in the document]
Comment (GordonSmith) [reply cmake_modules/commonSetup.cmake:847]: (like ICU)

**Comment by AttilaVamos** [comment]
In Centos Smoketest ESP crashes with Segmentation Fault.

**Comment by GordonSmith** [comment]
> In Centos Smoketest ESP crashes with Segmentation Fault.

What are your cmake configuration settings?

**Comment by GordonSmith** [comment]
@ghalliday / @AttilaVamos I have fixed the ESP not starting issue (was happening when WSSQL_SERVICE=ON and USE_JAVA=OFF)

File: cmake_modules/commonSetup.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghalliday) [reply cmake_modules/commonSetup.cmake:847]: I would assume USE_C11_REGEX is true, and remove the code that was in the false branches.

**Comment by AttilaVamos** [comment]
Regression Suite Setup failed on Roxie based on missing Java. 
```
Error:
      1. Test: aaajavalibrary.ecl
    	Error: eclcc: (0,0): Error C6003: C++ link error: cannot find -ljavaembed
    eclccserver: (0,0): Error C3000: Compile/Link failed for W20230306-132909-5 (see W20230306-132909-5.cc.log for details)
    2 error(s), 0 warning(s)
```

**Review by jakesmith — APPROVED** [review]
Looks ok afaict.

**Comment by AttilaVamos** [comment]
Still failing on missing Java in Roxie setup.

**Comment by GordonSmith** [comment]
@ghalliday don't pull yet, I think this would benefit from a "BUILD_ALL" option, which can then be reused in build_assets, smoketests etc.  to ensure consistent testing / releasing?

**Review by richardkchapman — APPROVED** [review]
I did not spot anything I disagreed with

**Comment by GordonSmith** [comment]
Converting to draft until I can confirm build_assets is also succeeding

**Review by ghalliday — COMMENTED** [review]
@GordonSmith what is the state of this - it is still a draft and has conflicts with the target branch

**Pull request closed without merging**